docs/types/value-types.rst

```diff
@@ -338,6 +338,12 @@ You can query the deployed code for any smart contract. Use ``.code`` to get the
 ``bytes memory``, which might be empty. Use ``.codehash`` to get the Keccak-256 hash of that code
 (as a ``bytes32``). Note that ``addr.codehash`` is cheaper than using ``keccak256(addr.code)``.
 
+.. warning::
+    The output of ``addr.codehash`` depends on the ``addr`` balance. If ``addr`` has no code deployed
+    and the balance is 0, ``addr.codehash`` will output 0. If the balance is greater than 0, then
+    ``addr.codehash`` will output ``keccak256("")``. See `EIP-1052 <https://eips.ethereum.org/EIPS/eip-1052>`_
+    and `EIP-161 <https://eips.ethereum.org/EIPS/eip-161>`_.
```

Comment:
I suggest clarifying that 0 is returned if the account is non-existent or empty, rather than solely mentioning a zero balance. An account is considered empty when it has *no code* and *zero nonce* and *zero balance* (note the `ands`), as defined by [EIP-161](https://eips.ethereum.org/EIPS/eip-161). Additionally, if the account has no code, the keccak256 hash of empty data (i.e., `keccak256("") == c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470`) is returned, as defined by [EIP-1052](https://eips.ethereum.org/EIPS/eip-1052).

Reply:
@r0qs thanks, good notes! I've made another commit, what do you think?

Reply:
Thanks, I suggest some adjusts. Sorry for the late reply ;)